### PR TITLE
feat: Add Meson as a build option

### PR DESCRIPTION
**Pull request by mwcampbell**
This can coexist with the current CMake build system. I don't know yet if we want to keep the current CMake build system around in the long term or switch completely to Meson. AccessKit doesn't yet actually support any targets that Meson doesn't support, but that may change when we implement the Android backend in the next few months.

I'm making this a draft because this branch is currently based on the standalone-repo branch, which isn't merged. When that branch is merged, I can rebase this branch and take this PR out of draft status.

This PR is derived from librsvg's Meson build system, and most of the original files didn't have a copyright or license header. librsvg is under the LGPL. So my interpretation is that in the worst case, this means that accesskit-c is under the LGPL when built with Meson, but not when built with CMake or directly with cargo. But I would like clarification on that from @amyspark, who did most of the Meson work for librsvg. Once that's clarified, I'll add some kind of attribution and copyright header to these files.

### CODE REVIEW

**Comment by mwcampbell** [comment]
One notable change compared to librsvg: I decided not to use cargo-c. Since Meson is building the shared library and .pc file, and installing all of the artifacts, I concluded that cargo-c wasn't accomplishing anything that cargo itself couldn't do.

**Comment by mwcampbell** [comment]
Thanks @amyspark. What I meant to ask, though, is whether this means that the LGPL applies to our library itself when built with Meson. I'm sorry I wasn't clear about that.

**Comment by amyspark** [comment]
@mwcampbell as I understand it (and from my POV as author of those files) the LGPL only applies to **my files** and changes you make to them e.g. extending the `makedef` script.

You're free to release accesskit-c under the license that suits you best and I have no intentions to encumber that choice.

**Comment by mwcampbell** [comment]
@amyspark Thank you. I'll add a copyright and license notice.

**Comment by mwcampbell** [comment]
@amyspark Is it under the LGPL 3.0 or 2.1? I only see LGPL 2.1 in librsvg. I hate to nit-pick, but I want to make sure I get this right. Thanks.

**Comment by amyspark** [comment]
@mwcampbell `LGPL-2.1-or-later` for all. I've fixed the comment I made earlier to be absolutely clear.

I got confused for a moment since one file (`makedef.py`) comes originally from another repo of ours: https://gitlab.freedesktop.org/gstreamer/meson-ports/ffmpeg/-/blob/a5bc5338f300324028b7051d4b280a7ccef67d64/compat/windows/makedef.py

**Comment by mwcampbell** [comment]
I forgot that this PR was still a draft. I also forgot to rebase it before taking it out of draft status. Done now.

**Comment by DataTriny** [comment]
I am seeing the following warnings when setting up the project on Windows:

```
meson.build:349: DEPRECATION: Project uses feature that was always broken, and is now deprecated since '1.3.0': str.format: Value other than strings, integers, bools, options, dictionaries and lists thereof..
...
meson.build:394: WARNING: Project targets '>= 1.2.0' but uses feature introduced in '1.3.0': "shared_library" keyword argument "vs_module_defs" of type CustomTargetIndex.
...
WARNING: Project specifies a minimum meson_version '>= 1.2.0' but uses features which were added in newer versions:
 * 1.3.0: {'"shared_library" keyword argument "vs_module_defs" of type CustomTargetIndex'}
WARNING: Broken features used:
 * 1.3.0: {'str.format: Value other than strings, integers, bools, options, dictionaries and lists thereof.'}
```

Unless we have no way around, it would be better to fix these right now.

I encounter errors when linking so I can't tell if this is fully working at the moment, need to investigate more.

**Comment by lu-zero** [comment]
I noticed that you have a branch that adds the cargo-c metadata, did you have any problem with using it? If not could you also add it so distributions that prefer cargo-c can use it as well?

**Comment by mwcampbell** [comment]
I abandoned cargo-c in favor of Meson because the latter was necessary to meet GTK's needs. But I guess adding cargo-c as a third option wouldn't add much maintenance in the long term.

**Comment by mwcampbell** [comment]
OK, I think I've got this fully working on all three major platforms, including both MSVC and MSYS2 on Windows.

@amyspark I'd also appreciate your review, since this is based on your scripts for librsvg.

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+        ]
+)
+
+host_system = host_machine.system()
+cc = meson.get_compiler('c')
+
+# MSRV - Minimum Supported Rust Version
+# If you change this, please update these:
+#   - the "rust-version" value in Cargo.toml
+msrv = '1.70.0'
```
Comment (DataTriny) [inline meson.build:31]:
```suggestion
msrv = '1.75.0'
```

**Review by DataTriny — CHANGES_REQUESTED** [review]

**Review by DataTriny — APPROVED** [review]

**Pull request merged**